shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -411,29 +410,38 @@ - (void)loadAOTData:(NSString*)assetsDir {
 }
 
 - (void)setViewController:(FlutterViewController*)controller {
-  if (_viewController != controller) {
+  if (_viewController == controller) {
+    // From nil to nil, or from non-nil to the same controller.
+    return;
+  }
+  if (_viewController == nil && controller != nil) {
```

Comment:
~You can simplify this since if `_viewController` is nil, at this point we know that `controller` is non-nil or they'd have been equal in the check above.~

Actually, regardless of the fact that we _can_ simplify these, I think from a readability point of view this is easier to reason about. I'd be tempted to add comments though just for easy scanning of the code.
```suggestion
  // From nil to non-nil.
  if (_viewController == nil && && controller != nil) {
```